logstash-core-event/lib/logstash/event.rb

```diff
@@ -245,16 +243,24 @@ def fields
     raise DeprecatedMethod
   end
 
+  def self.logger=(logger)
+    @@logger = logger
```

Comment:
I am not sure that thread safety is a *real* problem here:
- `@@logger` is only assigned once normally. We currently do not reassign loggers at runtime.
- that being said, `@@logger` could probably be initialized sooner instead of in the the first `#Event#logger` call
- in JRuby defining or modifying constants and class variables is thread safe

Are there any other concerns I missed?